app/controllers/users_controller.rb

```diff
@@ -1,2 +1,63 @@
 class UsersController < ApplicationController
+  def index
+    @users = User.all
+  end
+
+  def show
+    @user = get_user
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    if params.fetch(:user).fetch(:password) == params.fetch(:user).fetch(:password_confirmation)
+      @user = User.new(user_params)
+      if @user.save!
+        flash[:notice] = "User created!"
+        redirect_to :back
+      else
+        flash.now[:alert] = "All fields are required"
+        redirect_to :back
+      end
+    else
+      flash[:alert] = "Paswords must match"
+      redirect_to :back
+    end
+  end
+
+  def edit
+    @user = get_user
+  end
+
+  def updated
+    @user = get_user
+    if @user.update(user_params)
+      redirect_to @user
+    else
+      flash.now[:alert] = "There was an issue"
+      redirect_to :back
+    end
+  end
+
+  def destroy
+    @user = get_user
+
+    if @user.destroy
+      redirect_to :users_path
+    else
+      flash.now[:alert] = "There was an issue"
```

Comment:
redirect_to issues a new http request. Flash persists through the next http request, and flash.now will NOT persist through the next http request. Flash.now should mainly only be used when calling "render" since "render" does not issue a new http request. As it appears here, the user will never see the flash.now[:alert] message because it will be wiped clean as soon as the redirect_to issues a new http request.